common-arcana.lic

```diff
@@ -489,7 +489,7 @@ module DRCA
     pause 0.5 while UserVars.moons.empty?
   end
 
-  def cast_spell(data, settings, force_cambrinth = false)
+  def cast_spell(data, settings, force_cambrinth = false, wait_lambda = nil, cast_lambda = nil)
```

Comment:
💡 What if we consolidate to one lambda and it's method signature accept an extra argument to know what phase of spell casting we're in so it can do appropriate logic?

For example, the lambda signature would accept `(phase, spell_data, settings)` where `phase` is one of **before prep**, **after prep**, **before cast**, **after cast**.

**before prep** would occur right before line 522.
**after prep** would occur where it does now in the PR.
**before cast** would occur where it does now in the PR.
**after cast** would occur right after line 560.

Reply:
That's fine with me if that's what people would prefer. No strong feelings one way or the other. Only reason I did it this way was because I had two use cases that each needed one hook.

Anyone else have thoughts on the preferred interface?

Reply:
Changing the signature seems like it would be nice to avoid passing things like (...,nil, nil, ..., nil), but I'm not strongly opinionated about it.